ClosedXML/Extensions/FontBaseExtensions.cs

```diff
@@ -36,13 +68,39 @@ public static double GetWidth(this IXLFontBase fontBase, string text, Dictionary
             {
                 return 0;
             }
+            var systemSpecificWidthScalingFactor = GetCachedWidthCalibration(fontCache);
+            var width = SystemSpecificWidthCalculator(fontBase, text, fontCache, systemSpecificWidthScalingFactor);
 
-            var font = GetCachedFont(fontBase, fontCache);
-            var textWidth = GraphicsUtils.MeasureString(text, font.Typeface).Width;
+            return width < maxExcelColumnWidth ? width : maxExcelColumnWidth;
+        }
+
+        private static double GetCachedWidthCalibration(Dictionary<IXLFontBase, SKFont> fontCache)
+        {
+            if (CachedWidthCalibrationFactor.HasValue)
+                return CachedWidthCalibrationFactor.Value;
+
+            var calibratedValue = 36.535187641402715d;
+            var text = "Very Wide Column";
 
-            var width = (textWidth / 7d * 256 - 128 / 7) / 256;
-            width = Math.Round(width + 0.2, 2);
+            var xLFont = new XLFont
+            {
+                FontSize = 20,
+                FontName = "Verdana"
+            };
 
+            var SystemSpecificWidthOfKnownWidth = SystemSpecificWidthCalculator(xLFont, text, fontCache, 1);
+            CachedWidthCalibrationFactor = calibratedValue / SystemSpecificWidthOfKnownWidth;
+
+            return CachedWidthCalibrationFactor.Value;
+        }
+
+        private static double SystemSpecificWidthCalculator(IXLFontBase fontBase, string text, Dictionary<IXLFontBase, SKFont> fontCache, double systemSpecificScalingFactor)
+        {
+            var font = GetCachedFont(fontBase, fontCache);
+            var marginPoints = ((font.Size * 0.4) + 8) / 1.326;
```

Comment:
I honestly have no idea on how you came up with that. Is there any chance to make a reader understand what is going on here? The what and how is described, but not the why. This could be important if something breaks in the future.

Reply:
added some comments